wp-content/mu-plugins/assets/locale-switcher.js

```diff
@@ -16,26 +16,105 @@
 			app.$notice   = $( '.wp20-locale-notice' );
 			app.$container = $( '.navigation-top-menu-container' );
 			app.$outerContainer = $( '.wp20-locale-switcher-container' );
+			app.$switcherForm = $( '#wp20-locale-switcher-form' );
 
-			app.$switcher.selectWoo( {
-				language: app.locale,
-				dir: app.dir,
-				dropdownParent: app.$container,
-				width: 'auto',
-				dropdownAutoWidth : true
-			} );
+			const localeSelect = document.querySelector( '#wp20-locale-switcher' );
+
+
+
+			// todo enable a11yautocom here
+			accessibleAutocomplete.enhanceSelectElement( {
+				selectElement: localeSelect,
+				showAllValues: true,
+				// displayMenu: 'overlay' might be useful
+				// autoselect: false
+
+				// Submit the form.
+				onConfirm: function( localeName ) {
+					// Find the selected locale code.
+					// This is necessary because of https://github.com/alphagov/accessible-autocomplete/issues/387.
+					const selectedOption = Array.from( localeSelect.querySelectorAll( 'option' ) ).find( function( option ) {
+						return option.innerText === localeName;
+					} );
+
+					if ( selectedOption ) {
+						if ( app.locale !== selectedOption.value ) {
+							$( '#wp20-locale-switcher-select' ).val( selectedOption.value );
+							app.$switcherForm.submit();
+						}
+					}
+				},
+
+				dropdownArrow: function() {
+					// Intentionally empty because we'll create the arrow in CSS.
+					return '';
+				},
+
+				// defaultValue: function( value ) {
+				// 	console.log( 'defaultvalue', value );
+				// 	return 'hi';
+				// }
+
+				// . If your use case doesn't fit the above defaults, try reading the source and seeing if you can write your own.
+
+				// souce - use locale code too
+
+				// inputValue: 		function( value ) {
+				// 	console.log( 'inputValue', value );
+				// return value;
+				// },
+				// doesn't give locale code
+
+				// dropdownArrow: () => '' requires showallvalues true
+
+				// transleat "no results found" etc
+				/*
+				do this just like you've done other i18n strings
+				tStatusQueryTooShort: (minQueryLength) => '',
+				tStatusNoResults: () => '',
+				tStatusSelectedOption: (selectedOption, length) => '',
+				tStatusResults: () => ''
 
-			app.$switcher.on( 'change', function() {
-				$(this).parents( 'form' ).submit();
+				make sure escaped
+				*/
+
+				// { templates: { inputValue, suggestion } } to let search by endonym and english name
+					// make sure escaped
+
+
+				placeholder: 'Search languages...',
```

Comment:
Current UI. String needs translation.

![Screen Shot 2023-04-27 at 2 23 54 PM](https://user-images.githubusercontent.com/1017872/234743732-50a4fddf-fc8f-41a8-b1c9-937f4a4f8034.jpg)

Reply:
Ah maybe it should use the string here https://github.com/WordPress/wp20.wordpress.net/pull/93/files#diff-1368bd0d70766b64532fd35e145bb6e6cea9b47f3297d37027f10d2f8865e763R30

Reply:
Done in https://github.com/WordPress/wp20.wordpress.net/pull/93/commits/54d4216fdcbb7da69b8b38e14e36b53239afe91c

Reply:
Yeah, that looks like a good approach 👍🏻